c++/src/kj/compat/http.c++

```diff
@@ -1616,35 +1620,68 @@ kj::Own<kj::AsyncInputStream> HttpInputStreamImpl::getEntityBody(
     }
   }
 
+  // #2 deals with the CONNECT method which is handled separately.
+
+  // #3
   KJ_IF_MAYBE(te, headers.get(HttpHeaderId::TRANSFER_ENCODING)) {
     // TODO(someday): Support plugable transfer encodings? Or at least gzip?
     // TODO(someday): Support stacked transfer encodings, e.g. "gzip, chunked".
+
+    // NOTE: #3¶3 is ambiguous about what should happen if Transfer-Encoding and Content-Length are
+    //   both present. It says that Transfer-Encoding takes precedence, but also that the request
+    //   "ought to be handled as an error", and that proxies "MUST" drop the Content-Length before
+    //   forwarding. We ignore the vague "ought to" part and implement the other two. (The
+    //   dropping of Content-Length will happen naturally if/when the message is sent back out to
+    //   the network.)
     if (fastCaseCmp<'c','h','u','n','k','e','d'>(te->cStr())) {
+      // #3¶1
       return kj::heap<HttpChunkedEntityReader>(*this);
+    } else if (fastCaseCmp<'i','d','e','n','t','i','t','y'>(te->cStr())) {
```

Comment:
This is an interesting case. RFC 2616 referred to an "identity" transfer encoding, but then forgot to define it (it referred to a nonexistent "section 3.6.2"). The body length calculations use it, though, so we can determine the length of the body by the connection being closed -- we just don't technically know how to decode the body!

This is functionally equivalent to RFC 7230, which removed the "identity" transfer encoding. So per that RFC we ought to treat identity transfer encoding as any other non-chunked encoding that we don't support: i.e., it ends at connection close, and we don't know how to decode it.

(I'm happy interpreting it with the obvious meaning as you do here; researching this just gave me some chuckles.)

Reply:
Doh, didn't notice that "identity" was actually removed in 7230. Oh well, doesn't hurt to recognize it, though I suppose no one uses it since it's... not useful.